### PR TITLE
Allow specific users to send email from any address

### LINKED ISSUE #1096: Allow specific users to send email from any address
Before switching to Mailu I had postfix set up so that my admin user `admin@mydomain.dev` could impersonate / send email as any other user.

This was extremely useful for sending email programmatically from multiple sources, e.g. from a web app that needs to send email from `*@mydomain.dev` and several other domains like `*@admin.mydomain.dev`/`*@someotherdomain.dev`.

I suspect I can add something into `overrides/postfix.conf` to get this behavior back, but I'm not sure how to do it since I only want a specific user to be able to impersonate others, not all users.  Ideally there would be a UI page to configure the permitted sender addresses for each user, with sqlite-style wildcard support.

```ini
smtpd_client_restrictions =
  permit_mynetworks,
  check_sender_access ${podop}senderaccess,
  reject_non_fqdn_sender,
  reject_unknown_sender_domain,
  reject_unknown_recipient_domain,
  reject_unverified_recipient,
  permit

smtpd_relay_restrictions =
  permit_mynetworks,
  permit_sasl_authenticated,
  reject_unauth_destination
```

For now, given I'm the only one using the server and I need this behavior back ASAP, can I just remove the `check_sender_access ${podop}senderaccess,` line? Or is there a better long-term fix?

I also cant just use `RELAYNETS` with some whitelisted IPs becuase the servers doing the sending have dynamic IP addresses that often change, I need it to be whitelisted on a per-user basis not a per-sender-ip basis.

**Useful links:**
- https://github.com/Mailu/Mailu/blob/master/core/postfix/conf/main.cf
- http://www.postfix.org/ADDRESS_VERIFICATION_README.html
- http://www.postfix.org/postconf.5.html#check_sender_access

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

Feature

## What does this PR do?

Allow specific users to send email from any address using the WILDCARD_SENDERS configuration variable.

### Related issue(s)
- closes #1096

## Prerequistes
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [x] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by ghostwheel42** [comment]
What about making this configurable in the web interface?

**Comment by nextgens** [comment]
The diff will be a lot larger (db migration, HTML, ...) but sure we can do it. Is it something we want to restrict to admins (ie that only admins can grant?) or something the user can self-elect for (a I-know-what-I'm-doing flag)?

**Comment by nextgens** [comment]
@ghostwheel42 I am having issues setting up the dev environment properly (see #1577) ... and as such, I am unable to make it configurable through the GUI... currently I rebuild/redeploy the container as required to test my changes and this is way too time intensive for doing something like this...

As users are actively requesting the feature, I propose that we merge it as is and once the instructions on setting up a working dev environment are updated I will endeavour to revisit and implement a GUI for it.

**Comment by Diman0** [comment]
That sounds reasonable to me. After this PR is merged, we must not forget to create an issue for adding this GUI config.

**Review by ghostwheel42 — CHANGES_REQUESTED** [review]
whitespace needs to be remove when using WILDCARD_SENDERS env

File: docs/configuration.rst
Comment (Diman0) [inline docs/configuration.rst:40]: Could you clarify this further that you can only spoof existing email addresses? For example:

The ``WILDCARD_SENDERS`` setting is a comma delimited list of user email addresses that are allowed to send emails from any existing email address (spoofing the sender).

**Review by Diman0 — CHANGES_REQUESTED** [review]
When testing I could only spoof existing email addresses. Is that expected? 
If yes, then please clarify the documentation on this ( I already placed the relevant in-line comment).

The error when using a non-existing email address:
MTP Error (550): Failed to add recipient "user@domain.com" (5.1.0 <doesnotexist@emaildomain.com>: Sender address rejected: User unknown in virtual mailbox table)

File: core/admin/mailu/internal/views/postfix.py
Comment (nextgens) [reply core/admin/mailu/internal/views/postfix.py]: Good idea; done in 7252a73e118650fc0737e9df39e07c853af53ac0

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors ping

@mergifyio ping

**Comment by nextgens** [comment]
bors cancel
bors retry

**Comment by nextgens** [comment]
bors retry

**Comment by Diman0** [comment]
another attempt

bors try

**Comment by Diman0** [comment]
bors cancel

**Comment by Diman0** [comment]
bors try

**Comment by Diman0** [comment]
bors retry

**Comment by Diman0** [comment]
Hmm I see that BORS crashed (https://app.bors.tech/repositories/6693/log). 
I will built the images locally and test with whitespace in the variable.  After that I will approve and check if the bors r+ from mergify is picked up. If it still fails, I can always merge it manually.

**Review by Diman0 — DISMISSED** [review]
With the latest changes I cannot send any emails anymore. It does not matter if the env var WILDCARD_SENDERDS is set or is not set (commented). Sending any email results in the error:

SMTP Error (451): Failed to add recipient "recipient@recipient_domain.nl" (4.3.0 <sender@sender_domain.nl>: Temporary lookup failure).

Can you double check on your environment if you experience the same?

**Comment by nextgens** [comment]
It looks like I've screwed up the merge... I hate git.

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
@Diman0 give it another shot please, this works for me now

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors cancel

**Comment by Diman0** [comment]
bors is working again. I will review it this evening.

**Review by Diman0 — APPROVED** [review]
Looking good now. The test with wildcard variable with lots of spaces succeeded now.

**Comment by miroslavskela** [comment]
Hello,
I have problem with this setting. Have implemented everything in  /mailu/data/overrides, and also after that didn't work moved to /mailu/overrides, but still cant sent mail even as another user on same mail server, with same domain.Does anyone have some idea what could be problem. sender_logins and sasl_passwd are empty. Thank you!

**Pull request merged**
